core-services/src/main/java/org/zalando/nakadi/service/timeline/TimelinesFollowerNode.java

```diff
@@ -0,0 +1,157 @@
+        executorService.scheduleWithFixedDelay(this::periodicExposeSelfVersionSafe, 1, 1, TimeUnit.MINUTES);
+    }
+
+    @PreDestroy
+    public void destroyNode() {
+        log.info("Stopping node updates");
+        executorService.shutdownNow();
+    }
+
+    /**
+     * There may be a situation, when session was terminated, but we don't know about that/ignored this problem.
+     * In order to avoid this problem, every now and then follower node exposes it's version (in order to ensure that
+     * it's own node is still there).
+     * Unfortunately, in case if there are many nodes running, it may create additional pressure on zookeeper. In order
+     * to avoid this - exposure should not run too frequent (in case if zookeeper connection is not lost - this method
+     * is not needed at all)
+     */
+    private void periodicExposeSelfVersionSafe() {
+        try {
+            log.debug("Exposing self version {} to ensure that node is still there", currentVersion.getVersion());
+            timelinesZookeeper.exposeSelfVersion(currentVersion.getVersion());
+        } catch (final RuntimeException ex) {
+            log.warn("Failed to periodically expose self version", ex);
+        } catch (final InterruptedException ex) {
+            log.warn("Was interrupted while periodically exposing self version");
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    /**
+     * Periodically or by zookeeper event checks the version of timelines state in zookeeper. In case if it is changed
+     * applies some actions. Most of the time (when timelines are not created) method acts in read-only mode, not
+     * adding write pressure on zookeeper.
+     */
+    private void refreshVersionSafe() {
+        try {
+            log.debug("Running periodic refresh version");
+            refreshVersion();
+        } catch (final RuntimeException ex) {
+            // During the refresh something bad has happened. Usually that means that there are 2 possible problems:
+            // 1. Zookeeper communication problem.
+            // 2. Internal lock refresh problem.
+            // In any case - it is required to execute the periodic refresh again, and the only question - after which
+            // grace period it should be done.
+            // Right now as there is no experience with behavior - it makes sense to rely on the default grace period
+            // not to spam zookeeper with updates.
+            log.error("Unexpected exception caught, can't throw it further because of single-threaded executor", ex);
+        } catch (final InterruptedException ex) {
+            log.warn("Interrupted waiting for changes, probably time to die", ex);
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private void refreshVersion() throws RuntimeException, InterruptedException {
+        final TimelinesZookeeper.ZkVersionedLockedEventTypes zkData = timelinesZookeeper.getCurrentVersion(null);
```

Comment:
it is not clear what current version is, please, rename it eg: ``getLockedEventTypes`

Reply:
or `getState`